Extensions/xkit_patches.js

```diff
@@ -184,6 +184,16 @@ XKit.extensions.xkit_patches = new Object({
 
 	patches: {
 		"7.9.2": function() {
+			XKit.tools.cartesian_product = (items, current = []) => {
```

Comment:
I get that a cartesian product is a defined math term but I don't think anyone else is going to use this function without a plain english description and a few jsdoc type annotations